src/main/pacman/pacman_agent.py

```diff
@@ -0,0 +1,350 @@
+import random
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+from tensorflow.keras.layers import Dense, Dropout, Conv2D, MaxPooling2D, Activation, Flatten
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.optimizers import Adam
+
+
+# REWARDS = [] # Temporary. Only for plotting the output
+class Blob:
+    # Definitions related to each blob in the environment (walls, enemies, player, food).
+    def __init__(self, *args):
+
+        if len(args) == 1:
+            self.map_size = args[0]
+            self.x = np.random.randint(0, self.map_size)
+            self.y = np.random.randint(0, self.map_size)
+        elif len(args) == 3:
+            self.map_size = args[0]
+            self.x = args[1]
+            self.y = args[2]
+        else:
+            print("ERROR: Incorrect Number of Args. Should be 1 or 3.")
+            assert False
+
+    def __str__(self):
+        return f"Blob ({self.x}, {self.y})"
+
+    def __sub__(self, other):
+        return (self.x - other.x, self.y - other.y)
+
+    def __eq__(self, other):
+        return self.x == other.x and self.y == other.y
+
+    def action(self, choice, walls):
+        '''
+        Gives us 9 total movement options. (0,1,2,3,4,5,6,7,8)
+        '''
+        if choice == 0:
+            self.move(walls, x=1, y=1)
+        elif choice == 1:
+            self.move(walls, x=-1, y=-1)
+        elif choice == 2:
+            self.move(walls, x=-1, y=1)
+        elif choice == 3:
+            self.move(walls, x=1, y=-1)
+
+        elif choice == 4:
+            self.move(walls, x=1, y=0)
+        elif choice == 5:
+            self.move(walls, x=-1, y=0)
+
+        elif choice == 6:
+            self.move(walls, x=0, y=1)
+        elif choice == 7:
+            self.move(walls, x=0, y=-1)
+
+    def move(self, walls, x=False, y=False):
```

Comment:
Did you added more choices, if yes Then will it require any update in move function too? is it already done.